packages/pyright-internal/src/common/configOptions.ts

```diff
@@ -1282,6 +1289,7 @@ export function getStrictDiagnosticRuleSet(): DiagnosticRuleSet {
         reportUnannotatedClassAttribute: 'none',
         reportIncompatibleUnannotatedOverride: 'none',
         reportInvalidAbstractMethod: 'none',
+        reportSelfClsDefault: 'error',
```

Comment:
i don't turn on any basedpyright specific rules in any of the diagnostic rulesets that exist upstream
```suggestion
        reportSelfClsDefault: 'none',
```